R/query.R

```diff
@@ -229,10 +229,43 @@ batches <- function(access_token = check_access_token())
 #' Used to authenticate level of access to data. 
 #' By default, checks if already authenticated through [`get_access_token`] and 
 #' if not then does not perform authentication
+#' @param gwasglue Returns a gwasglue2 SummarySet object  (if `gwasglue = TRUE`).  Only one GWAS id can be queried at a time. See [gwasglue2::create_dataset()].Default = `FALSE`.
 #'
 #' @export
-#' @return Dataframe
-associations <- function(variants, id, proxies=1, r2=0.8, align_alleles=1, palindromes=1, maf_threshold = 0.3, access_token=check_access_token())
+#' @return Dataframe. If `gwasglue = TRUE` then returns a gwasglue2 object.
+associations <- function(variants, id, proxies=1, r2=0.8, align_alleles=1, palindromes=1, maf_threshold = 0.3, access_token=check_access_token(), gwasglue=FALSE){
+
+	# Query specific variants from specific GWAS using associations_query internal function (old version)
+	out <- associations_query(variants=variants, id=id, proxies=proxies, r2=r2, align_alleles=align_alleles, palindromes=palindromes, maf_threshold=maf_threshold, access_token=access_token)
+		
+	if(isTRUE(gwasglue))
+	{
+		# check if it is a tibble (trying to avoid loading the tibble package)
+		if(any(class(out)=="tbl_df")){
+    		# output gwasglue2 SummarySet object
+			if(id %>% length() != 1){
+				stop("Only one GWAS ID can be queried at a time when using `gwasglue = TRUE`.")
+			} else {
+				# create gwasglue2 metadata 
+				m <- gwasglue2::create_metadata(gwasinfo(id))
+				# create gwasglue2 SummarySet object
+				s <- out %>% 
+					gwasglue2::create_summaryset(metadata=m, qc = TRUE) %>%
+					return()
```

Comment:
Because it returns a S4 object, it needs to be in a variable.

`s <- associations(variants="7:105561135-105563135", id=c("ieu-a-2"),gwasglue=TRUE)`

If you call `s` it will display the summary of the gwasglue2 `SummarySet` object. The package then has several methods to query and acess the information stored in the different slots of `s`.